qiskit/providers/backend.py

```diff
@@ -229,3 +239,392 @@ class can handle either situation.
             Job: The job object for the run
         """
         pass
+
+
+class QubitProperties:
+    """A representation of the properties of a qubit on a backend.
+
+    This class provides the optional properties that a backend can provide for
+    a qubit. These represent the set of qubit properties that Qiskit can
+    currently work with if present. Howeverm if your backend provides additional
+    properties of qubits you should subclass this to add additional custom
+    attributes for those custom/additional properties provided by the backend.
+    """
+
+    __slots = ("t1", "t2", "frequency")
```

Comment:
Are we doing something weird here, or is this a `__slots__` typo?  Also, given that above you suggested that classes might want to inherit from this: are we _absolutely_ sure that there will never be a use-case for multiple inheritance from two different types of `QubitProperties`?  I guess so, but just bringing it up because only one immediate parent can define nonempty `__slots__`.

```suggestion
    __slots__ = ("t1", "t2", "frequency")
```

Reply:
Heh, yeah it's just a pretty awful typo. I don't think there would ever be a multiple inheritance use case. It's basically more if your implementing a backend and you have more qubit properties to report adding a subclass for it is the recommended path to do that. It shouldn't interfere with using slots here as we want these properties to be fast to access because they're the ones we can use in terra today.